client/discovery/consul.go

```diff
@@ -0,0 +1,82 @@
+package discovery
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+)
+
+// ConsulDiscovery is a back-end for service discovery which can be used
+// to populate a local Consul agent with service information. Because
+// Consul already has information about the local node, some shortcuts
+// can be taken in this back-end. Specifically, the IP address of the Nomad
+// agent does not need to be used, because Consul has this information
+// already and may even be configured to expose services on an alternate
+// advertise address.
```

Comment:
I may be misreading this but I'm not sure this is a safe assumption to make. Since nomad may be managing tasks across multiple IPs and those services will be bound to a specific interface, the IP is actually important. Even if nomad or consul is listening on a particular IP, tasks are not necessarily registered on the same IP.